src/main/java/com/eternalcode/parcellockers/user/UserRepositoryJdbcImpl.java

```diff
@@ -0,0 +1,96 @@
+        return CompletableFuture.runAsync(() ->
+                this.jdbcConnectionProvider.executeUpdate("INSERT INTO `users` (`uuid`, `name`) VALUES (" + user.getUuid() + "," + user.getName() + ") ON DUPLICATE KEY UPDATE `name` = ?"));
+    }
+
+    @Override
+    public Optional<User> findByName(String name) {
+        try (ResultSet resultSet = this.jdbcConnectionProvider.executeQuery("SELECT * FROM `users` WHERE `name` = ? LIMIT 1".replace("?", name))) {
+            Set<Parcel> parcelSet = this.parcelRepository.findAll();
+            Set<UUID> userParcels = new HashSet<>();
+            while (resultSet.next()) {
+                for (Parcel parcel : parcelSet) {
+                    if (parcel.getSender().equals(UUID.fromString(resultSet.getString("uuid")))) {
+                        userParcels.add(parcel.getUuid());
+                    }
+                }
+            }
+            return Optional.of(new User(UUID.fromString(resultSet.getString("uuid")), resultSet.getString("name"), userParcels));
+        } catch (SQLException exception) {
+            throw new RuntimeException(exception);
+        }
+    }
+
+    @Override
+    public Optional<User> findByUuid(UUID uuid) {
+        try (ResultSet resultSet = this.jdbcConnectionProvider.executeQuery("SELECT * FROM `users` WHERE `uuid` = ? LIMIT 1".replace("?", uuid.toString()))) {
+            Set<Parcel> parcelSet = this.parcelRepository.findAll();
+            Set<UUID> userParcels = new HashSet<>();
+            while (resultSet.next()) {
+                for (Parcel parcel : parcelSet) {
+                    if (parcel.getSender().equals(UUID.fromString(resultSet.getString("uuid")))) {
+                        userParcels.add(parcel.getUuid());
+                    }
+                }
+            }
+            return Optional.of(new User(UUID.fromString(resultSet.getString("uuid")), resultSet.getString("name"), userParcels));
+        } catch (SQLException exception) {
+            throw new RuntimeException(exception);
+        }
+    }
+
+    @Override
+    public List<User> findAll() {
+        Set<Parcel> parcelSet = this.parcelRepository.findAll();
+        List<User> users = new ArrayList<>();
+
+        try (ResultSet resultSet = this.jdbcConnectionProvider.executeQuery("SELECT * FROM `users`")) {
+            while (resultSet.next()) {
+                Set<UUID> userParcels = new HashSet<>();
+                for (Parcel parcel : parcelSet) {
+                    if (parcel.getSender().equals(UUID.fromString(resultSet.getString("uuid")))) {
+                        userParcels.add(parcel.getUuid());
+                    }
+                }
+                users.add(new User(UUID.fromString(resultSet.getString("uuid")), resultSet.getString("name"), userParcels));
+            }
+        } catch (SQLException exception) {
+            throw new RuntimeException(exception);
+        }
+        return users;
+    }
```

Comment:
to co wyżej, usera możesz wydzielić do osobnego obiektu by było to bardziej czytelne